### PR TITLE
[Security] [Guard] Improve comment with working example

### PR DESCRIPTION
**Pull request by pasdeloup**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Improve the code given in the comment to make it work.
If you follow blindly the previous example (as I did) and never return null, you're stucked in an infinite redirection loop on your login page.

### CODE REVIEW

File: src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php [hunk 1]
```diff
@@ -39,11 +39,16 @@
      * Whatever value you return here will be passed to getUser() and checkCredentials()
      *
      * For example, for a form login, you might:
-     *
-     *      return array(
-     *          'username' => $request->request->get('_username'),
-     *          'password' => $request->request->get('_password'),
-     *      );
+     * 
+     *      if($request->request->has('_username')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php]: CS needs a fix: space after `if`, and `else` on the same line as `}` (but is the `else` required?)

Comment (pasdeloup) [inline src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php]: Sorry again for the CS, I should really forget my old habits and get used to real PSR-2...

For the `else` I want to emphasis on the fact that you should not forget to return null when you don't have credentials.

File: src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php [hunk 2]
```diff
@@ -39,11 +39,15 @@
...
-     *      );
+     * 
+     *      if ($request->request->has('_username')) {
+     *          return array(
+     *              'username' => $request->request->get('_username'),
+     *              'password' => $request->request->get('_password'),
+     *          );
+     *      } else {
+     *          return;
+     *      }
```
Comment (iltar) [inline src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php]: The else is not not necessary, the if statement around it is sufficient enough.

Question though, why would this create an infinite redirect?

Comment (pasdeloup) [inline src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php]: In the previous example, you return an array, even if no query is sent.

So when you arrive on your login/ form page, you send this array, so Guard tries to authenticate you.
As it fails to to it, you are redirected to... the login page again.
And so on.

As I said to @nicolas-grekas , the else is not necessary but the point is to show you should not forget this null case.

**Comment by fabpot** [comment]
Thank you @pasdeloup.

**Pull request closed without merging**